force-app/main/default/classes/ServiceDeliveryController.cls

```diff
@@ -9,6 +9,34 @@
 
 public with sharing class ServiceDeliveryController {
     public static ServiceService service = new ServiceService();
+    @TestVisible
+    private static ServiceDeliverySelector deliverySelector = new ServiceDeliverySelector();
+
+    @TestVisible
+    private static ServiceDeliveryDomain deliveryDomain = new ServiceDeliveryDomain();
+
+    @AuraEnabled
+    public static Integer deleteServiceDeliveriesForSession(Id sessionId) {
+        try {
+            List<ServiceDelivery__c> deliveries = deliverySelector.getServiceDeliveriesForSession(
+                sessionId
+            );
+
+            deliveryDomain.deleteServiceDeliveries(deliveries);
+            return deliveries.size();
+        } catch (Exception e) {
+            throw Util.getAuraHandledException(e);
+        }
+    }
+
+    @AuraEnabled(cacheable=true)
+    public static Integer getNumberOfServiceDeliveriesForSession(Id sessionId) {
+        try {
+            return deliverySelector.getNumberOfServiceDeliveriesForSession(sessionId);
```

Comment:
I think that is definitely worth considering, and I am definitely caught between two approaches here. I was hoping to avoid sending/storing the list of deliveries to the LWC component since that "leaks" additional information to the front-end, and the COUNT() query should be pretty cheap from a resource perspective.

On the other hand, this method includes a little additional apex/SOQL and there's a corner case where some other user or automation adds a delivery in between when the modal loads and the first user hits delete - that could cause the advertised number of deleted deliveries to be different from the actual number, but it wouldn't leave a lingering service delivery behind.

Maybe I will check with PM super fast...?

Reply:
Given that Claire likes the current behavior - deleting all deliveries even if the count has shifted while the modal is open - I'm inclined to make a light push to keep this as is. The separate COUNT() query should be more efficient and doesn't lock records, etc, and the overall approach avoids sending more than necessary information to the front end.